crates/algokit_utils/src/transactions/composer.rs

```diff
@@ -1156,40 +1335,44 @@ impl Composer {
             }
         }
 
-        let transactions = self
+        let build_results = self
             .build_transactions(&suggested_params, &default_validity_window, group_analysis)
             .await?;
 
-        let transactions_with_signers = self.gather_signers(transactions);
+        let transactions_with_signers = self.gather_signers(build_results);
 
         self.built_group = Some(transactions_with_signers?);
         Ok(self.built_group.as_ref().unwrap())
     }
 
     fn gather_signers(
         &self,
-        transactions: Vec<Transaction>,
+        build_results: Vec<BuildTransactionResult>,
     ) -> Result<Vec<TransactionWithSigner>, ComposerError> {
-        transactions
+        build_results
             .into_iter()
             .enumerate()
-            .map(|(group_index, txn)| {
-                let ctxn = &self.transactions[group_index];
-                let common_params = ctxn.common_params();
-                let signer = if let Some(transaction_signer) = common_params.signer {
-                    transaction_signer
-                } else {
-                    let sender_address = txn.header().sender.clone();
-                    self.get_signer(sender_address.clone())
-                        .ok_or(ComposerError::SigningError(format!(
-                            "No signer found for address: {}",
-                            sender_address
-                        )))?
-                };
-                Ok(TransactionWithSigner {
-                    transaction: txn,
-                    signer,
-                })
+            .map(|(group_index, build_result)| match build_result {
+                BuildTransactionResult::TransactionWithSigner(tx_with_signer) => Ok(tx_with_signer),
+                BuildTransactionResult::Transaction(txn) => {
+                    let ctxn = &self.transactions[group_index];
+                    let common_params = ctxn.common_params();
+                    let signer = if let Some(transaction_signer) = common_params.signer {
```

Comment:
Something about this feels weird. This transaction has a signer, but is not a transaction with signer.